File: lab01/ejer01.py
def solicitar_numero(nombre_variable):
    while True:
        entrada = input(f"Ingrese el valor de {nombre_variable} (puede ser entero o decimal): ")
        try:
            valor = float(entrada)
            return valor
        except ValueError:
            print(f"Error: '{entrada}' no es un numero valido. Intente nuevamente.")

def calcular_area(base, altura):
    return base * altura

def main():
    print("=== Calculo del área de un rectángulo ===")
    base = solicitar_numero("la base")
    altura = solicitar_numero("la altura")
    
    area = calcular_area(base, altura)
    
    print("\n--- Resultado ---")
    print(f"Base ingresada: {base}")
    print(f"Altura ingresada: {altura}")
    print(f"Area del rectangulo: {area}")

if __name__ == "__main__":
    main()

File: lab02/src/main/java/com/miapp/ejer01.java
package lab02.src.main.java.com.miapp;

import java.util.Scanner;

public class ejer01 {
    /* 
    public static double solicitarNumero(String nombreVariable, Scanner scanner) {
        double valor;
        while (true) {
            try {
                System.out.print("Ingrese el valor de " + nombreVariable + " (puede ser entero o decimal): ");
                valor = Double.parseDouble(scanner.nextLine());
                if (valor <= 0) {
                    System.out.println("Error: El valor debe ser mayor que cero. Intente nuevamente.");
                    continue;
                }
                return valor;
            } catch (NumberFormatException e) {
                System.out.println("Error: No es un número válido. Intente nuevamente.");
            }
        }
    }
        */
    public static double calcularArea(double base, double altura) {
        System.out.println("/////PRUEBA///// base: " + base + " altura : "+ altura + " area calculada: " + (base * altura));
        return base * altura;
    }
    /* 
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("=== Cálculo del área de un rectángulo ===");
        double base = solicitarNumero("la base", scanner);
        double altura = solicitarNumero("la altura", scanner);
        double area = calcularArea(base, altura);

        System.out.println("\n--- Resultado ---");
        System.out.printf("Base ingresada: %.2f%n", base);
        System.out.printf("Altura ingresada: %.2f%n", altura);
        System.out.printf("Área del rectángulo: %.2f%n", area);
    }
        */
}
